namer/namer.cc

```diff
@@ -618,7 +618,14 @@ class NameInserter {
     }
 
     unique_ptr<ast::Assign> fillAssign(core::MutableContext ctx, unique_ptr<ast::Assign> asgn) {
-        // TODO(nelhage): forbid dynamic constant definition
+
+        // forbid dynamic constant definition
+        if (!ctx.owner.data(ctx)->isClass()) {
+            if (auto e = ctx.state.beginError(asgn->loc, core::errors::Namer::DynamicConstant)) {
```

Comment:
maybe use different error code?

Reply:
Should I make a new one? This one seemed appropriate, and wasn't used anywhere else.

Reply:
OK, I made the new code `DynamicConstantAssignment` and switched to that here.